internal/exec/exec.go

```diff
@@ -293,10 +297,9 @@ func (r *Request) executeWithDirectives(ctx context.Context, f *fieldToExec, arg
 	currResolver := resolver
 
 	for i := n - 1; i >= 0; i-- {
-		i := i
-		directive := f.field.Directives[i]
+		d := f.field.Directives[i]
 
-		if visitor, ok := r.Visitors[directive.Name.Name]; ok {
+		if visitor, ok := r.Visitors[d.Name.Name]; ok {
```

Comment:
Maybe we need to instantiate per resolver instance of the visitor here instead of reusing the same one from the map.

Reply:
Ah, `r` in this context is actually `*Request` this means that all visitors are reused across the schema. However, if we add the visitor instance to each `f *fieldToExec`, just like its `PackedArgs`, it might as well have a `Visitors` field. Then your idea with the type-safe arguments directly mapped to struct fields and it will be safe in a multi-threaded environment.

Reply:
Also some directives are repeable For example imagine this schema:
```graphql
schema {
    ...
    mutation: Mutation
}
directive @hasRole(role: String!) repeatable ON FIELD_DEFINITION

type Mutation {
    updatePost(id: ID, patch: UpdatePostInput!): UpdatePostResult! @hasRole(role: Journalist) @hasRole(role: Editor)
    publishPost(id: ID): PublishPostResult! @hasRole(role: Editor)
    ...
}
```
In the `updatePost` resolver example the `hasRole` is even added twice to the same field with a different argument. So the users would get an error if they are not a journalist and not an editor at the same time.